### ISSUE TITLE
Publish-Item to use Sitecore Publishing Service

### ISSUE THREAD

**Issue by dejanstamenov** [SitecorePowerShell/Console#1013]
### Expected Behavior

**Publish-Item** command to use the Sitecore Publishing service, if the Sitecore instance is configured to use this service for publishing instead the "old" in-built Publishing.

### Actual Behavior

**Publish-Item** command still uses the "old" in-build Publishing when using PowerShell script. The Sitecore Publishing Service is configured on my instance, and is working properly. But I cannot invoke it from PowerShell script using the SPE, or I am missing something here?

### Steps to Reproduce the Problem

I have the following script (simplified):

```
[CmdletBinding()]
param (
    [Parameter(Mandatory=$false)]
    [System.String] $UnicornSyncItemsJson = 'C:\Users\XXX\Documents\PS-Sync-Items.json'
)
begin {
    Import-Module -Name SPE
    [PSCustomObject] $JsonFileData = Get-Content $UnicornSyncItemsJson | Out-String | ConvertFrom-Json
}
process {
    $ExecutionBlock = {
        $SessionJsonData = $Using:JsonFileData
        [Sitecore.Data.Items.BaseItem[]] $SyncedItems = @()


        $SessionJsonData.SyncItems | ForEach-Object {
            [Sitecore.Data.Items.BaseItem] $Item = Get-Item -Path 'master:' -ID $_.Id

            if ([System.String]::IsNullOrEmpty($Item.ItemPath)) {
                Write-Verbose -Message ('Unable to identify item: "{0}"!' -F $_.Id)
            }
            else {
                $SyncedItems += $Item
                Write-Verbose -Message ('Item: "{0}" added to the queue!' -F $_.Id)
            }
        }

        if ($SyncedItems.Count -gt 0) {
            $SyncedItems | Publish-Item -PublishMode Smart
        }
    }

    $Session = New-ScriptSession -Username 'admin' -Password 'b' -ConnectionUri 'http://sitecore'
    Invoke-RemoteScript -Session $Session -ScriptBlock $ExecutionBlock -Verbose
    Stop-ScriptSession -Session $Session
}
```

In short, I have a JSON file containing all the IDs from the items that were synced with Unicorn. I need to publish these items from PowerShell using the Sitecore Publishing Service. The script itself works, but is using the "old" in-built Publishing from Sitecore. If I am missing something, please do let me know.
Using **SPE 4.7** with **Sitecore 8.2 Update 3**.

 - [**YES**] Tested issue with clean install of Sitecore and the latest available version of SPE.
 
 - [**NO**] Asked questions on the Sitecore Slack Chat channel.
 
 - [**YES**] Reviewed questions and answers on the Sitecore Stack Exchange.

**Comment by michaellwest** [comment]
@dejanstamenov  can you give this a try?

```powershell
$database = Get-Database -Name "master"

$item = Get-Item -Path "master:" -ID "{110D559F-DEA5-42EA-9C1C-8A5DF7E70EF9}"

$targets = @()
foreach($publishingTarget in [Sitecore.Publishing.PublishManager]::GetPublishingTargets($database)) {
    $targets += Get-Database -Name $publishingTarget[[Sitecore.FieldIDs]::PublishingTargetDatabase]
}

$languages = [Sitecore.Data.Managers.LanguageManager]::GetLanguages($database)
[Sitecore.Publishing.PublishManager]::PublishItem($item,$targets,$languages,$true,$true,$true)
```

![image](https://user-images.githubusercontent.com/933163/50533419-99c79980-0aee-11e9-84fd-261ffdcf3e2e.png)

**Comment by dejanstamenov** [comment]
Hi @michaellwest,

Thank you for the guidance. It works fine with the snippet you have provided. I have altered it a bit and will provide it below for further reference, in case someone else needs it.

```
[CmdletBinding()]
param (
    [Parameter(Mandatory=$false)]
    [System.String] $UnicornSyncItemsJson = 'C:\Users\XXX\Documents\PS-Sync-Items.json'
)
begin {
    Import-Module -Name SPE # Location - C:\Users\XXX\Documents\WindowsPowerShell\Modules\

    [PSCustomObject] $JsonFileData = Get-Content $UnicornSyncItemsJson | Out-String | ConvertFrom-Json
}
process {
    $ExecutionBlock = {
        [Sitecore.Data.DefaultDatabase] $masterDatabase = Get-Database -Name "master"

        [System.Object[]] $publishingTargets = @()
        foreach ($target in [Sitecore.Publishing.PublishManager]::GetPublishingTargets($masterDatabase)) {
            $publishingTargets += Get-Database -Name $target[[Sitecore.FieldIDs]::PublishingTargetDatabase]
        }

        [Sitecore.Collections.LanguageCollection] $languages = [Sitecore.Data.Managers.LanguageManager]::GetLanguages($masterDatabase)

        $SessionJsonData = $Using:JsonFileData
        [Sitecore.Data.Items.Item[]] $SyncedItems = @()

        $SessionJsonData.SyncItems | ForEach-Object {
            [Sitecore.Data.Items.Item] $Item = Get-Item -Path 'master:' -ID $_.Id

            if ([System.String]::IsNullOrEmpty($Item.ItemPath)) {
                Write-Verbose -Message ('Unable to identify item: "{0}"!' -F $_.Id)
            }
            else {
                $SyncedItems += $Item
                Write-Verbose -Message ('Item: "{0}" added to the queue!' -F $_.Id)
            }
        }

        if ($SyncedItems.Count -gt 0) {
            Write-Verbose -Message ('There are {0} items that need to be synced!' -F $SyncedItems.Count)
            $SyncedItems | ForEach-Object {
                Write-Verbose -Message ('Sync item: "{0}".' -F $_.Id)
                [Sitecore.Publishing.PublishManager]::PublishItem($_, $publishingTargets, $languages, $true, $true, $true)
            }
        }
    }

    $Session = New-ScriptSession -Username 'admin' -Password 'b' -ConnectionUri 'http://sitecore'
    Invoke-RemoteScript -Session $Session -ScriptBlock $ExecutionBlock -Verbose
    Stop-ScriptSession -Session $Session
}
```

Have also done some checks in the database, especially the tables related to the Publishing Service. The jobs, manifests and all the steps are properly generated, which proves out the fact that this code works with the Publishing Service.
